pds4indextools/pds4_create_xml_index.py

```diff
@@ -198,271 +701,781 @@ def update_nillable_elements_from_xsd_file(xsd_file, nillable_elements_info):
+    valid_choices (list): A list of valid choices that each value in the comma-separated
+                          list must be part of.
+
+    Returns:
+    list: A list of values parsed from the input string.
+
+    Raises:
+    argparse.ArgumentTypeError: If any value in the comma-separated list is not in the
+                                valid choices.
+    """
+    values = arg_value.split(',')
+    for value in values:
+        if value not in valid_choices:
+            raise argparse.ArgumentTypeError(f"Invalid choice: '{value}' "
+                                             f"(choose from {valid_choices})")
+    return values
+
+
+def generate_unique_filename(base_name):
+    """
+    Generate a unique filename by appending a number to the base_name if it already
+    exists.
+
+    Args:
+    base_name (str): The base name of the file including the extension.
+
+    Returns:
+    str: A unique filename that does not already exist in the current directory. The
+        filename is generated by appending a number to the base name before the file
+        extension.
+    """
+    counter = 1
+    new_filename = base_name
+    base, extension = os.path.splitext(base_name)
+
+    while os.path.exists(new_filename):
+        new_filename = f"{base}{counter}{extension}"
+        counter += 1
+
+    return new_filename
+
+
+def main(cmd_line=None):
+    # FIXIT: CHANGE THIS LINK AFTER MERGED TO MAIN
+    parser = argparse.ArgumentParser(
+        epilog="For more details, please visit the online documentation at: "
+        "https://github.com/SETI/rms-pds4indextools/blob/es-unit_tests/README.md"
+        )
+
+    valid_add_extra_file_info = ['lid', 'filename', 'filepath', 'bundle_lid', 'bundle']
+
+    index_file_generation = parser.add_argument_group('Index File Generation')
+    index_file_generation.add_argument('directorypath', type=str,
+                                       help='The path to the directory containing the '
+                                            'bundleset, bundle, or collection you wish '
+                                            'to scrape')
+
+    index_file_generation.add_argument('patterns', type=str, nargs='+',
+                                       help='The glob pattern(s) for the files you wish '
+                                            'to index. They may include wildcards '
+                                            'like *, ?, and **. If supplying multiple '
+                                            'patterns, separate with spaces. Surround '
+                                            'each pattern with quotes.')
+
+    index_file_generation.add_argument('--output-index-file', type=str,
+                                       help='Specify the location and filename of the '
+                                            'index file.')
+
+    index_file_generation.add_argument(
+        '--add-extra-file-info',
```

Comment:
One more nice thing you can do with argparse is add the `metavar` option. This is what shows up in `--help` instead of the option name. For example, currently we get:
```
  --add-extra-file-info ADD_EXTRA_FILE_INFO
                        Add additional columns to the final index file. If specifying multiple column names, supply them as separate arguments separated by spaces.
```
but "ADD_EXTRA_FILE_INFO" is not really helpful, and nowhere does it say what the options are. So you can add to the `add_argument` definition something like:
```
metavar='COMMA_SEPARATED_COLUMN_NAMES'
```
and then put in the help string something about what the valid options are.

I would do this for everything that takes an argument. For example:
```
--sort-by SORT_BY
```
would make more sense as:
```
--sort-by COMMA_SEPARATED_HEADER_NAME(s)
```
and
```
--generate-label GENERATE_LABEL
```
would make more sense as:
```
--generate-label {Product_Ancillary,Product_Metadata_Supplemental}
```